core/common/src/main/java/alluxio/ParameterizedPropertyKey.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import alluxio.exception.ExceptionMessage;
+
+import com.google.common.base.Objects;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A set of parameterized property keys that can generate the property name given parameters.
+ * E.g., * {@code ParameterizedPropertyKey.MASTER_TIERED_STORE_GLOBAL_LEVEL_ALIAS.format(0)}
+ */
+@ThreadSafe
+public enum ParameterizedPropertyKey {
```

Comment:
given its name, I would expect this class to extend the `PropertyKey` class ... I think this class is closer to `PropertyKeyFormat`